bi/pkg/cluster/kind/control_server.go

```diff
@@ -0,0 +1,107 @@
+package kind
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"net/http"
+	"strings"
+	"time"
+)
+
+// ControlServerChecker verifies that the control server is accessible
+type ControlServerChecker struct {
+	baseURL string
+}
+
+func NewControlServerChecker(baseURL string) *ControlServerChecker {
+	return &ControlServerChecker{
+		baseURL: baseURL,
+	}
+}
+
+func (csc *ControlServerChecker) WaitForControlServer(ctx context.Context, timeout time.Duration) error {
+	if strings.Contains(csc.baseURL, "localhost") {
+		slog.Info("Skipping control server check for localhost - VPN connection required")
+		return fmt.Errorf("localhost check skipped - VPN connection required")
+	}
+	
+	slog.Info("Waiting for control server to become accessible", 
+		slog.String("url", csc.baseURL),
+		slog.Duration("timeout", timeout))
+	
+	ctx, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+	
+	ticker := time.NewTicker(5 * time.Second)
+	defer ticker.Stop()
+	
+	client := &http.Client{
+		Timeout: 10 * time.Second,
+	}
```

Comment:
When setting up the HTTP client, we'll need to add the WireGuard transport here. 

See an example here: https://github.com/noisysockets/noisysockets/blob/9912d4c4a67b8b539f8bfa3f81b2a5dc72c8d89c/examples/router/main.go#L84

or here: https://github.com/batteries-included/batteries-included/blob/58a8e70f899e9df03d47df029dba546054193e8f/bi/pkg/kube/kube.go#L81-L103